jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java

```diff
@@ -212,11 +213,10 @@ private ManagedSelector chooseSelector(SelectableChannel channel)
         // The ++ increment here is not atomic, but it does not matter,
         // so long as the value changes sometimes, then connections will
         // be distributed over the available selectors.
-        long s = _selectorIndex++;
-        int index = (int)(s % getSelectorCount());
+        int index = _selectorIndex.updateAndGet(i->(i+1)%_selectors.length);
         ManagedSelector candidate2 = _selectors[index];
 
-        if (candidate1 == null || candidate1.size() >= candidate2.size() * 2)
+        if (candidate1 == null || candidate1.size() > candidate2.size())
```

Comment:
Now that I look at this, the logic for `candidate1` is A) a server-side only and B) does not hold because we are not pinning selector threads to cores. With EWYK this is even more true, so I think it should be completely removed.

Reply:
I agree... stand by....